internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -483,9 +483,25 @@ func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 		return diag.FromErr(err)
 	}
 
-	cluster, _, err := connV2.ClustersApi.CreateCluster(ctx, projectID, params).Execute()
-	if err != nil {
-		return diag.FromErr(fmt.Errorf(errorCreate, err))
+	var clusterName string
+	var clusterID string
+	var err error
+	if isUsingOldAPISchemaStructure(d) {
```

Comment:
pass by improvement, I believe this terminology aligns more with our docs (both public and internal)
```suggestion
	if isUsingOldShardingConfiguration(d) {
```

Reply:
Done